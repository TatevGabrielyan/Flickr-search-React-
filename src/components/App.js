import React, { Fragment, PureComponent } from 'react';
import { Provider } from 'react-redux';

import configureStore from '../config/store';
import Title from './Title';
import Search from './Search';
import Gallery from './Gallery';
import Detail from './Detail';
const store = configureStore();

class App extends PureComponent {
  render() {
    return (
      <Provider store={store}>
        <Fragment>
          <Title />
          <Search />
          <Gallery />
          <Detail />
        </Fragment>
      </Provider>
    );
  }
}

export default App;

