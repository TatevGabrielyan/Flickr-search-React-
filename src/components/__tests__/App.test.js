import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mockComponent } from '../../utils/testUtils';

import App from '../App';

jest.mock('../Gallery', () => {
  return props => mockComponent('Gallery', props);
});

jest.mock('../Search', () => {
  return props => mockComponent('Search', props);
});

jest.mock('../Title', () => {
  return props => mockComponent('Title', props);
});

jest.mock('../Detail', () => {
  return props => mockComponent('Detail', props);
});
jest.mock('../DragAndDrop', ()=>{
  return props=> mockComponent('DragAndDrop', props)
})

describe('App component', () => {
  const component = <App />;

  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Snapshot matchs', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
