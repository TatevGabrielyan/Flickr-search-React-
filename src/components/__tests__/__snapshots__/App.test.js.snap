// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Snapshot matchs 1`] = `
Array [
  <code
    name="Title"
    props={Object {}}
  />,
  <code
    name="Search"
    props={Object {}}
  />,
  <code
    name="Gallery"
    props={Object {}}
  />,
  <code
    name="Detail"
    props={Object {}}
  />,
  <code
  name="DragAndDrop"
  props={Object{}}
  />,
]
`;
